### ISSUE TITLE
[Google Sheets Extension] `SERPAPI_RESULT` formula returning `#ERROR!` for some countries

### ISSUE THREAD

**Issue by martin-serpapi** [serpapi/public-roadmap#1684]
A user reported that the `SERPAPI_RESULT` formula returns `#ERROR!` when some countries are selected in the `Locale` settings in Google Sheets. Below is an example using `Locale=Argentina`:

![image](https://github.com/serpapi/public-roadmap/assets/144151895/a3ab0d27-8b8d-477b-9e27-46b19178f780)

The same formula returns results when a different country, Australia for example, is selected in the `File -> Settings -> General -> Locale` settings:

![image](https://github.com/serpapi/public-roadmap/assets/144151895/3aeee4a7-5b61-4099-a166-e0f05be8db47)

`SERPAPI_RESULT` formula:

`=SERPAPI_RESULT("engine=google&q=Coffee&location=Austin, Texas, United States&google_domain=google.com&gl=us&hl=en", "organic_results.0.title")`

[Front](https://app.frontapp.com/inboxes/teams/10643996/inbox/open/102807425820) | [intercom](https://app.intercom.com/a/inbox/yzck75jr/inbox/shared/all/conversation/102910000067843?view=List)

**Comment by dsojevic** [comment]
I haven't been able to reproduce this with any of the first 30 countries available in the locale selection.

@martin-serpapi, are you able to reproduce this consistently?

**Comment by marm123** [comment]
@dsojevic it's more than likely the issue with the separator. For `locale=Argentina` the correct separator is `;`, and if the user tries to force the formula with query and selector separated with `,`, the Error will be raised:

<img width="781" alt="image" src="https://github.com/user-attachments/assets/135c9a51-c7c4-4c94-9538-f8d2096fdb64">

But if you change the separator, everything works:
<img width="780" alt="image" src="https://github.com/user-attachments/assets/ab1825ef-daf0-48d0-947c-8224df39e716">

It works correctly with `locale=Australia`, because the separator for that `locale` is `,`, and it looks like the user is trying to force the formula with this separator.

I'm not sure if that makes sense; let me know if not.

@martin-serpapi, can you confirm this might be the case? If yes, I think we can close as not planned, as it's not the issue with anything other than the syntax.

**Comment by dsojevic** [comment]
> @dsojevic it's more than likely the issue with the separator. For `locale=Argentina` the correct separator is `;`, and if the user tries to force the formula with query and selector separated with `,`, the Error will be raised:

You're absolutely correct here, @marm123! I was using the the semi-colon as a separator and it functioned across all locales... I would wager the error occurs when using a comma as an argument separator in locales that have a comma as a decimal separator.

While this is definitely a result of Google Sheets behaviour itself, I think we can probably adjust the code we provide in our docs and in the examples in the addon to use a semi-colon instead of a comma to separate the arguments and then it would work regardless of locale selected.

**Comment by marm123** [comment]
Yeah, I agree. It would save a lot of headaches for some users if we could provide a sample that works regardless of the locale. To be honest, I thought our samples already adjusted automatically to the locale, as I remember testing that in the past, but it looks like they don't - formulas are not adjusted to my current locale. I must have misremembered.

I'll open a separate issue for that tomorrow, as this one is not exactly accurate with what we discussed and might be misleading to other users.
